### PR TITLE
Get tick data

### PR DESCRIPTION
**Pull request by irenepapaspyratos**
By now only the Lambda for hourly trigger without terraform to keep the PR smaller :)

### CODE REVIEW

File: data-crawl/functions.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import variables
+import os
+import urllib.request
+import urllib.error
+from datetime import datetime
+import boto3
+
+def preparation(symbol_storagepath: str = variables.BASEPATH_SYMBOL, logfilepath: str = variables.DUCA_LOGFILE_PATH):
```
Comment (fabianschmauder) [inline data-crawl/functions.py]: can you find a better name?

File: data-crawl/functions.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+def load_ticks_hour_dukascopy(year:int, month:int, day:int, hour:int, symbol:str, symbol_storagepath: str = variables.BASEPATH_SYMBOL, logfilepath: str = variables.DUCA_LOGFILE_PATH):
+    url = f'{variables.DUCA_APIBASE}/{symbol.upper()}/{year:04d}/{(month-1):02d}/{day:02d}/{hour:02d}h_ticks.bi5'
+    filename = f'{year:04d}-{(month-1):02d}-{day:02d}-{hour:02d}h_ticks.bi5'
+    for i in range(5):
+        try:
+            urllib.request.urlretrieve(url, f'{symbol_storagepath}{filename}')
+        except urllib.error.HTTPError as e:
+            with open(logfilepath, 'a') as file:
+                file.write('ERROR: ' + str(e.status) + ' ' + url)
```
Comment (fabianschmauder) [inline data-crawl/functions.py]: do you really need to save it to a file?
File access is slow. If the date is not too huge then keep it in memory. Or write multiple files at once

File: data-crawl/functions.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        except Exception as e:
+            with open(logfilepath, 'a') as file:
+                file.write('EXCEPTION: ' + str(e) + ' ' + url)
+
+def create_s3_client():
+    return boto3.client('s3')
```
Comment (fabianschmauder) [inline data-crawl/functions.py:36]: why not have it as field?

Comment (fabianschmauder) [reply data-crawl/functions.py:36]: if you use a function you will always create a new client object

File: data-crawl/functions.py (hunk 1 above)
Comment (irenepapaspyratos) [reply data-crawl/functions.py]: I will try. 🤷🏻‍♀️

File: data-crawl/functions.py (hunk 3 above)
Comment (irenepapaspyratos) [reply data-crawl/functions.py:36]: Your second comment is clear. This function will later be called once at the beginning of the handler.
Your first comment: HÄ???

Comment (fabianschmauder) [reply data-crawl/functions.py:36]: ok i mean you can also define a variable at the top. but ok

**Pull request merged**